src/custom/utils/trade.ts

```diff
@@ -231,22 +233,25 @@ type OrderCancellationParams = {
 }
 
 export async function sendOrderCancellation(params: OrderCancellationParams): Promise<void> {
-  const { orderId, account, chainId, signer, cancelPendingOrder } = params
+  const { orderId, chainId, signer, cancelPendingOrder } = params
 
   const { signature, signingScheme } = await signOrderCancellation(orderId, chainId, signer)
 
-  await sendSignedOrderCancellation({
-    chainId,
-    owner: account,
-    cancellation: { orderUid: orderId, signature, signingScheme },
-  })
+  await orderBookApi.sendSignedOrderCancellation(
+    orderId,
+    {
+      signature,
+      signingScheme,
+    },
+    { chainId }
+  )
 
   cancelPendingOrder({ chainId, id: orderId })
 }
 
 export async function hasTrades(chainId: ChainId, address: string): Promise<boolean> {
   const [trades, profileData] = await Promise.all([
-    getTrades({ chainId, owner: address, limit: 1 }),
+    orderBookApi.getTrades({ owner: address }, { chainId }),
```

Comment:
Be aware the limit!
Simple optimization because this call only wants to know whether the account has any trades.
Having 1 is enough, no need to get the default page size (100? 1000?)

Reply:
Yeah :) But it doesn't work, because there is no pagination on the backend.
https://cowservices.slack.com/archives/C0375NV72SC/p1677250746103339

Reply:
:O

I was sure there was!
Was probably thinking about the orders endpoint.